### ISSUE TITLE
angular - Generate pages: Error: Specified module 'home/home.module.ts' does not exist.

### ISSUE THREAD

**Issue by PieterjanDeClippel** [nrwl/nx#10009]
## Offending command

    ng g module home --module app --route home

## Current Behavior

I'm getting an error when generating a page inside an application:

    Specified module 'home/home.module.ts' does not exist.

## Steps to Reproduce

Create the application
-------------------------

```
npx create-nx-workspace
> pages-test
> angular
> pages-demo
> scss
> No
code pages-test
```

Create the AppRoutingModule (app-routing.module.ts)
---------------------------------

```
import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';

const routes: Routes = [];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule { }
```

Import the `AppRoutingModule` in the `AppModule`

Generate a page
------------------

```
cd apps/pages-demo/src/app
ng g module home --module app --route home
```

Result:

```
Error: Specified module 'home/home.module.ts' does not exist.
Looked in the following directories:
    /apps/pages-demo/src/app/home/home.module.ts
    /apps/pages-demo/src/app/home
    /apps/pages-demo/src/app
    /apps/pages-demo/src
    /apps/pages-demo
    /apps
    at findModuleFromOptions (C:\repos\pages-test\node_modules\@schematics\angular\utility\find-module.js:48:15)
    at C:\repos\pages-test\node_modules\@schematics\angular\component\index.js:109:66        
Error: Specified module 'home/home.module.ts' does not exist.
    /apps/pages-demo/src/app/home/home.module.ts
    /apps/pages-demo/src/app/home
    /apps/pages-demo/src/app
    /apps/pages-demo/src
    /apps/pages-demo
    /apps
    at findModuleFromOptions (C:\repos\pages-test\node_modules\@schematics\angular\utility\find-module.js:48:15)
    at C:\repos\pages-test\node_modules\@schematics\angular\component\index.js:109:66
Specified module 'home/home.module.ts' does not exist.
Looked in the following directories:
    /apps/pages-demo/src/app/home/home.module.ts
    /apps/pages-demo/src/app/home
    /apps/pages-demo/src/app
    /apps/pages-demo/src
    /apps/pages-demo
    /apps
```

### Environment

- NX: 14.0.3

```
Angular CLI: 13.3.3
Node: 16.13.1
Package Manager: npm 8.1.2
OS: win32 x64

Angular: undefined
...

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.1303.3 (cli-only)
@angular-devkit/core         13.2.3
@angular-devkit/schematics   13.2.3
@schematics/angular          13.2.3
```

### Debug NX schematics

[Here's what I did to be able to debug nx commands](https://www.youtube.com/watch?v=OGV4R0cPRPc)

**Comment by AgentEnder** [comment]
Can you run `nx report` and list the output here? Thanks!

**Comment by PieterjanDeClippel** [comment]
```
PS C:\repos\pages-test\apps\pages-demo\src\app> ng g module home --module app --route home 
Debugger listening on ws://127.0.0.1:52135/e96c1dba-5b8e-43b6-8520-84f7d26f4c2a
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
Error: Specified module 'home/home.module.ts' does not exist.
Looked in the following directories:
    /apps/pages-demo/src/app/home/home.module.ts
    /apps/pages-demo/src/app/home
    /apps/pages-demo/src/app
    /apps/pages-demo/src
    /apps/pages-demo
    /apps
    at findModuleFromOptions (C:\repos\pages-test\node_modules\@schematics\angular\utility\find-module.js:48:15)
    at C:\repos\pages-test\node_modules\@schematics\angular\component\index.js:109:66        
Error: Specified module 'home/home.module.ts' does not exist.
    /apps/pages-demo/src/app/home/home.module.ts
    /apps/pages-demo/src/app/home
    /apps/pages-demo/src/app
    /apps/pages-demo/src
    /apps/pages-demo
    /apps
    at findModuleFromOptions (C:\repos\pages-test\node_modules\@schematics\angular\utility\find-module.js:48:15)
    at C:\repos\pages-test\node_modules\@schematics\angular\component\index.js:109:66
Specified module 'home/home.module.ts' does not exist.
Looked in the following directories:
    /apps/pages-demo/src/app/home/home.module.ts
    /apps/pages-demo/src/app/home
    /apps/pages-demo/src/app
    /apps/pages-demo/src
    /apps/pages-demo
    /apps
Waiting for the debugger to disconnect...
PS C:\repos\pages-test\apps\pages-demo\src\app> ng g module home --module app --route home
Error: Specified module 'home/home.module.ts' does not exist.
Looked in the following directories:
    /apps/pages-demo/src/app/home/home.module.ts
    /apps/pages-demo/src/app/home
    /apps/pages-demo/src/app
    /apps/pages-demo/src
    /apps/pages-demo
    /apps
    at findModuleFromOptions (C:\repos\pages-test\node_modules\@schematics\angular\utility\find-module.js:48:15)
    at C:\repos\pages-test\node_modules\@schematics\angular\component\index.js:109:66
Error: Specified module 'home/home.module.ts' does not exist.
Looked in the following directories:
    /apps/pages-demo/src/app/home/home.module.ts
    /apps/pages-demo/src/app/home
    /apps/pages-demo/src/app
    /apps/pages-demo/src
    /apps/pages-demo
    /apps
    at findModuleFromOptions (C:\repos\pages-test\node_modules\@schematics\angular\utility\find-module.js:48:15)
    at C:\repos\pages-test\node_modules\@schematics\angular\component\index.js:109:66
Specified module 'home/home.module.ts' does not exist.
Looked in the following directories:
    /apps/pages-demo/src/app/home/home.module.ts
    /apps/pages-demo/src/app/home
    /apps/pages-demo/src/app
    /apps/pages-demo/src
    /apps/pages-demo
    /apps



PS C:\repos\pages-test\apps\pages-demo\src\app> nx report

 >  NX   Report complete - copy this into the issue template

   Node : 16.13.1
   OS   : win32 x64
   npm  : 8.1.2

   nx : 13.10.3
   @nrwl/angular : 13.10.3
   @nrwl/cypress : 13.10.3
   @nrwl/detox : Not Found
   @nrwl/devkit : 13.10.3
   @nrwl/eslint-plugin-nx : 13.10.3
   @nrwl/express : Not Found
   @nrwl/jest : 13.10.3
   @nrwl/js : Not Found
   @nrwl/linter : 13.10.3
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 13.10.3
   @nrwl/web : Not Found
   @nrwl/workspace : 13.10.3
   typescript : 4.6.3
   rxjs : 7.4.0
   ---------------------------------------
   Community plugins:
```

**Comment by PieterjanDeClippel** [comment]
OS: Windows 11

**Comment by PieterjanDeClippel** [comment]
Can you show me where the schematic resides to generate a module? I see one for a component, directive, ... but not for a module

**Comment by leosvelperez** [comment]
@PieterjanDeClippel thanks for reporting this!

I haven't been able to reproduce this on macOS, I'm guessing this is a Windows issue with paths. The schematic for generating a module comes from the `@schematics/angular` package. Our `@nrwl/angular` package just extends the schematic collection from that one and therefore, all schematics from that package that we don't overwrite are executed from there. From the stack trace you shared, you can see the error being thrown at `C:\repos\pages-test\node_modules\@schematics\angular\component\index.js:109:66`.

I'm trying to set up a Windows VM here to troubleshoot this, but I've been having issues with it. It does seem to be an issue in the Angular CLI repo rather than an Nx issue given we don't own that schematic, but I'll keep this open to get to the bottom of it.

**Comment by PieterjanDeClippel** [comment]
The odd thing is that I'm not having this problem in an angular workspace, only an NX workspace

**Comment by leosvelperez** [comment]
I see. Thanks for confirming that, it might be something on our adapter layer then. I'll try to get that Windows VM running to reproduce it and troubleshoot it.

**Comment by PieterjanDeClippel** [comment]
@AgentEnder 
What do we need to do to get this new version containing this fix?

**Comment by leosvelperez** [comment]
@PieterjanDeClippel the fix was released in [v14.1.0](https://github.com/nrwl/nx/releases/tag/14.1.0).
